x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlSearchIT.java

```diff
@@ -94,7 +97,15 @@ public void testAllTypesWithRequestToOldNodes() throws Exception {
             columns -> {
                 columns.add(columnInfo("geo_point_field", "geo_point"));
                 columns.add(columnInfo("float_field", "float"));
-                columns.add(columnInfo("half_float_field", "half_float"));
+                /*
+                 * In 7.12.x we got full float precision from half floats
+                 * back from the fields API. When the cluster is mixed with
+                 * that version we don't fetch the half float because we
+                 * can't make any assertions about it.
+                 */
+                if (false == halfFloatMightReturnFullFloatPrecision) {
```

Comment:
@astefan it looked like the easiest way to get this backport landed was to disable pulling half_float if the you might get back more precision than the test expects. If we could put a `Matcher` in the map I could assert that it is "close". But we can't put `Matcher`s in the map.

I *think* this means that for 7.12 we won't be checking half floats. But we check them before 7.12 and after it.

Reply:
Er. Looks like I made a mistake. I'll reply once I've sorted it.

Reply:
OK. I think I've sorted it. There are a couple of more versions where we skip half float - but only when we send the request to the upgraded node.